src/H5FD.c

```diff
@@ -1142,6 +1144,146 @@ H5FDfree(H5FD_t *file, H5FD_mem_t type, hid_t dxpl_id, haddr_t addr, hsize_t siz
     FUNC_LEAVE_API(ret_value)
 } /* end H5FDfree() */
 
+/*----------------------------------------------------------------------------
+ * Function:    H5FD__error_matches
+ *
+ * Purpose:     Checks if the most recent frame of the error stack have a
+ *              major and minor numbers that match the provided arguments.
+ *
+ * Return:      TRUE if arguments match the most recent error, FALSE otherwise
+ *
+ *----------------------------------------------------------------------------
+ */
+static hbool_t
+H5FD__error_matches(hid_t major, hid_t minor)
```

Comment:
Agreed with @derobins on the package vs. private separation. This kind of a method is likely something that can (and should) be part of the H5E private code and then we wouldn't need to bring in H5Epkg.h. There should probably also be a bit more sanity checking here, like making sure the error stack actually has any messages on it before accessing `slot[estack->nused-1]`.